### PR TITLE
Bump dxvk and split it into multiple extensions based on versions number

### PR DESCRIPTION
**Pull request by Lctrs**
Depends on #17.

### CODE REVIEW

**Comment by julianrichen** [comment]
Hello,

The branch name will actually need to be `3.0` or the Sdk & Platform won't pick it up because it's [version is searching for 3.0](https://github.com/winepak/winepak-sdk-images/blob/master/org.winepak.Sdk.yml#L43).

I'm wondering if we should ship each version as 

```
org.winepak.Platform.Extension.dxvk-59
org.winepak.Platform.Extension.dxvk-60
org.winepak.Platform.Extension.dxvk-61
```

Like the flatpak teams does Nvidia drivers. Does dxvk regress enough that's it's an issue that we need each version or can we just ship it as `org.winepak.Platform.Extension.dxvk` with always the latest version?

**Comment by Lctrs** [comment]
DXVK bump regularly the minimum required wine version. So we might need a different branch for each version in case a game is not compatible with the new wine version.

**Comment by julianrichen** [comment]
Alright, let's change the version-ing to `*.Extension.dxvk-XX` so things don't break. That's kind of the point of winepak anyway.

Changes:

- `org.winepak.Platform.Extension.dxvk.yml` -> `org.winepak.Platform.Extension.dxvk-060.yml`
- branch number from `0.60` -> `3.0`
- for each source's `dest-filename` in the scripts module let's rename `dxvk-install` to `dxvk-install64` and `dxvk-install32` for 64 bit and 32 bit respectively so it matches the other modules.

**Comment by Lctrs** [comment]
@julianrichen done.

File: extension/org.winepak.Platform.Extension.dxvk-61.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      - type: archive
+        url: https://github.com/doitsujin/dxvk/releases/download/v0.61/dxvk-0.61.tar.gz
+        sha256: d04388f026dc0d8b276b08f7db74fb3556cbbc8f762401eb5ef52629ee39ded1
+
+  - name: scripts
+    buildsystem: simple
+    build-commands:
+      - install dxvk-install* ${FLATPAK_DEST}/bin
+    sources:
+      - type: script
```
Comment (julianrichen) [inline extension/org.winepak.Platform.Extension.dxvk-61.yml]: I would like to split these into two scripts `dxvk-install64` & `dxvk-install64-wow64` so they are similar to the others. Both use `wine64` but the `dxvk-install64-wow64` script only installs the `*/syswow64` stuff.

File: extension/org.winepak.Platform.Extension.dxvk-61.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        dest-filename: dxvk-install64
+        commands:
+          - if [ -z "$WINEPREFIX" ] ; then
+          - '    echo "No wine prefix set or is empty, abort."'
+          - '    exit 1'
+          - fi
+          -
+          - EXTENSION_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/../" && pwd )"
+          -
+          - mkdir -p "${WINEPREFIX}/dosdevices/c:/windows/system32/"
```
Comment (julianrichen) [inline extension/org.winepak.Platform.Extension.dxvk-61.yml]: I looked more into dxvk, the following code won't actually install & enable it. Instead we need to use the `setup_dxvk.sh`.

```
dlls/x64/setup_dxvk.sh -y -q # For dxvk-install64
dlls/x32/setup_dxvk.sh -y -q # For dxvk-install64-wow64 & dxvk-install32
```

Need to re-include the `setup_dxvk.sh` script that was stripped out above.

File: extension/org.winepak.Platform.Extension.dxvk-61.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+appstream-compose: false
+
+modules:
+  - name: dxvk
+    buildsystem: simple
+    build-commands:
+      - install -d ${FLATPAK_DEST}/bin/
+      - install -d ${FLATPAK_DEST}/dlls/x32/
+      - install -d ${FLATPAK_DEST}/dlls/x64/
+      - install -D x32/d3d11.dll ${FLATPAK_DEST}/dlls/x32/
```
Comment (julianrichen) [inline extension/org.winepak.Platform.Extension.dxvk-61.yml]: Install the whole `x32` & `x64` so we also get the `setup_dxvk.sh` file (reason explained below).

File: extension/org.winepak.Platform.Extension.dxvk-61.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        dest-filename: dxvk-install32
+        commands:
+          - if [ -z "$WINEPREFIX" ] ; then
+          - '    "No wine prefix set or is empty, abort."'
+          - '    exit 1'
+          - fi
+          -
+          - EXTENSION_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/../" && pwd )"
+          -
+          - mkdir -p "${WINEPREFIX}/dosdevices/c:/windows/system32/"
```
Comment (julianrichen) [inline extension/org.winepak.Platform.Extension.dxvk-61.yml]: Same as above.

**Comment by julianrichen** [comment]
I tested the code, it's not currently installing because we need to use the `setup_dxvk.sh` script. I really shouldn't have merged the original pr in a hurry before I left for vacation. At least I didn't push it to the repo...

File: extension/org.winepak.Platform.Extension.dxvk-61.yml (hunk 2 above)
Comment (Lctrs) [reply extension/org.winepak.Platform.Extension.dxvk-61.yml]: I tried to use these scripts while experimenting with the Fortnite installer with no luck. I'll look more into it tomorrow.

**Comment by Lctrs** [comment]
@julianrichen I think I finally managed to have a working install.

**Comment by Lctrs** [comment]
@julianrichen since dxvk seems to follow semver, we could be on the safe side by doing an extension per minor version (eg. `dvxk-05x`, `dxvk-06x`, etc.).

**Comment by julianrichen** [comment]
Cool, doing semver would also lower maintenance cost on our end. I'll hopefully be able to look into it more after the weekend and finally merge, builder, & push to the repo.

**Comment by LeandroStanger** [comment]
Update DXVK to version 0.64 https://github.com/doitsujin/dxvk/releases/tag/v0.64

**Comment by Lctrs** [comment]
Will have time to do it on monday probably.

**Comment by julianrichen** [comment]
@Lctrs just tested all the changes and dxvk seems to install fine, great job! However, unable to test with a real game. See https://github.com/winepak/applications/pull/71#issuecomment-410751319

**Comment by LeandroStanger** [comment]
Update DXVK to version 0.70 https://github.com/doitsujin/dxvk/releases/tag/v0.70

**Comment by Lctrs** [comment]
Hi @LeandroStanger.

I'm moving in a new apartment so I won't be able to do it until a few days.

**Comment by LeandroStanger** [comment]
Update DXVK to version 0.71 https://github.com/doitsujin/dxvk/releases/tag/v0.71

**Comment by julianrichen** [comment]
I managed to play with this a bit today, tried to get WoW to use dxvk but had problems rendering :( Either way the scripts look good and the files install fine. Only issue I see is that 05x & 06x still have reference to the patch file that was used before the better 07x version was push. Can you remove those and I'll merge.

Still working on the Sdk redesign branch but this should work in both the new and current version so I can push it to the server now.

**Pull request closed without merging**